framework/decode/dx12_replay_consumer_base.cpp

```diff
@@ -1986,14 +1986,17 @@ HRESULT Dx12ReplayConsumerBase::OverrideGetBuffer(DxObjectInfo*                r
         auto swapchain_info = GetExtraInfo<DxgiSwapchainInfo>(replay_object_info);
         if (swapchain_info != nullptr)
         {
-            if (swapchain_info->images[buffer] == nullptr)
+            if ((swapchain_info->image_ids.size() > buffer) &&
```

Comment:
Is `swapchain_info->image_ids.size() <= buffer` an exceptional condition that should have an assert, or considered so unlikely we shouldn't bother checking?

Reply:
The check `(swapchain_info->image_ids.size() > buffer)` was added here to handle potentially bad application input for the buffer index in the call to `GetBuffer()`, but because `SUCCEEDED(replay_result)` is checked above the buffer index should have already been validated. I've switched instead to an assert.